tremor-value/src/value/iter.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022, The Tremor Team
```

Comment:
Could we move that to a contribution to simd-json's value_trait it seems generally useful

Reply:
Do you think the logic employed for `IntoIterator` and the `.array_iter()` function make sense as they are?

Reply:
`array_iter` is 👍 into iter from the general idea too, I'm just a bit thinking if the trait makes sense as it would equally make sense to have `IntoIter` for a object and not an array but I'm not sure how to achieve that nicely to be honest

Reply:
I am also having a hard time coming up with a good iterator interface. I think I am going to ditch the whole `iter` stuff for now. array::flatten also works without this.